Userland/Libraries/LibJS/Runtime/TypedArray.cpp

```diff
@@ -347,6 +347,78 @@ ThrowCompletionOr<TypedArrayBase*> typed_array_create(GlobalObject& global_objec
     return &typed_array;
 }
 
+// 1.2.2.1.2 CompareTypedArrayElements ( x, y, comparefn, buffer ), https://tc39.es/proposal-change-array-by-copy/#sec-comparetypedarrayelements
+ThrowCompletionOr<double> compare_typed_array_elements(GlobalObject& global_object, Value x, Value y, FunctionObject* comparefn, ArrayBuffer& buffer)
+{
+    auto& vm = global_object.vm();
+    // 1. Assert: Both Type(x) and Type(y) are Number or both are BigInt.
+    VERIFY(((x.is_number() && y.is_number()) || (x.is_bigint() && y.is_bigint())));
+
+    // 2. If comparefn is not undefined, then
+    if (comparefn != nullptr) {
+        // a. Let v be ? ToNumber(? Call(comparefn, undefined, « x, y »)).
+        auto value = TRY(call(global_object, comparefn, js_undefined(), x, y));
+        auto value_number = TRY(value.to_number(global_object));
+
+        // b. If IsDetachedBuffer(buffer) is true, throw a TypeError exception.
+        if (buffer.is_detached())
+            return vm.throw_completion<TypeError>(global_object, ErrorType::DetachedArrayBuffer);
+
+        //  c. If v is NaN, return +0𝔽.
+        if (value_number.is_nan())
+            return 0;
+
+        // d. Return v.
+        return value_number.as_double();
+    }
+
+    // 3. If x and y are both NaN, return +0𝔽.
+    if (x.is_nan() && y.is_nan())
+        return 0;
+
+    // 4. If x is NaN, return 1𝔽.
+    if (x.is_nan())
+        return 1;
+
+    // 5. If y is NaN, return -1𝔽.
+    if (y.is_nan())
+        return -1;
+    auto x_value = x.is_bigint() ? x.as_bigint().big_integer() : x.as_double();
```

Comment:
This is not going to work like this, the `x.as_double()` causes an implicit cast to `i32` to invoke the implicit `SignedBigInteger(i32 x)` ctor.

![image](https://user-images.githubusercontent.com/19366641/176371149-c0dc3d49-5600-42b0-adb3-7715cad85ad0.png)

Reply:
Given this is only needed in two branches below, you can probably get rid of `x/y_value` and do

```cpp
if (x.is_bigint()
        ? (x.as_bigint().big_integer() < y.as_bigint().big_integer())
        : (x.as_double() < y.as_double()))
```

and

```cpp
if (x.is_bigint()
        ? (x.as_bigint().big_integer() > y.as_bigint().big_integer())
        : (x.as_double() > y.as_double()))
```